fileMapper.js

```diff
@@ -225,6 +226,21 @@ function recurseFolder(node, callback, filepath = '', depth = 0) {
   return depth === 0 ? undefined : __break;
 }
 
+// Performs recursive walk of HubSpot project
+async function doRemoteWalk(accountId, currPath, fileList = []) {
+  const contents = await getDirectoryContentsByPath(accountId, currPath);
```

Comment:
There's another [endpoint](https://tools.hubteam.com/api-catalog/services/ContentFileMapperService/v1/spec/internal#operations-Metadata-post-%2Fcontent%2Ffilemapper%2Fv1%2Fmeta%2Fbatch) dev infra added a few months ago that lets you batch these requests and do log(n) network calls here instead of n

It might make more sense to ask dev infra about adding an endpoint to do the walk server side though instead to avoid needing to reload the relevant memory on the server for each "layer". I was planning to ask for the same thing to make local previewing initial upload more efficient.

Reply:
> There's another [endpoint](https://tools.hubteam.com/api-catalog/services/ContentFileMapperService/v1/spec/internal#operations-Metadata-post-%2Fcontent%2Ffilemapper%2Fv1%2Fmeta%2Fbatch) dev infra added a few months ago that lets you batch these requests and do log(n) network calls here instead of n

Ooh neat! Didn't know about this. 

> It might make more sense to ask dev infra about adding an endpoint to do the walk server side though instead to avoid needing to reload the relevant memory on the server for each "layer". I was planning to ask for the same thing to make local previewing initial upload more efficient.

Yeah, I agree. Probably a good idea to just wait on the whole `--remove` feature until this is added as the request part  takes a really long time and ~doubles the time the command takes to run.

Reply:
Filed an [issue](https://git.hubteam.com/HubSpot/CMS-Developer-Infrastructure/issues/904)
!